app/src/org/commcare/android/crypt/CryptUtil.java

```diff
@@ -78,58 +80,44 @@ private static Cipher decodingCipher(String password) throws NoSuchAlgorithmExce
     }
 
     public static byte[] decrypt(byte[] input, Cipher cipher) {
-
         try {
             return cipher.doFinal(input);
-        } catch (IllegalBlockSizeException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        } catch (BadPaddingException e) {
-            // TODO Auto-generated catch block
+        } catch (IllegalBlockSizeException | BadPaddingException e) {
             e.printStackTrace();
         }
         return null;
-//        
-//        ByteArrayInputStream bis = new ByteArrayInputStream(input);
-//        CipherInputStream cis = new CipherInputStream(bis, cipher);
-//        
-//        ByteArrayOutputStream bos = new ByteArrayOutputStream();
-//        
-//        AndroidStreamUtil.writeFromInputToOutput(new BufferedInputStream(cis), bos);
-//        
-//        return bos.toByteArray();
     }
 
     public static byte[] wrapKey(byte[] secretKey, String password) {
+        return wrapByteArrayWithString(secretKey, password);
+    }
+
+    public static byte[] wrapByteArrayWithString(byte[] bytes, String wrappingString) {
         try {
-            //SecretKeySpec spec = (SecretKeySpec)SecretKeyFactory.getInstance("AES").getKeySpec(key, javax.crypto.spec.SecretKeySpec.class);
-            return encrypt(secretKey, encodingCipher(password));
-        } catch (InvalidKeySpecException e) {
-            throw new RuntimeException(e);
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
-        } catch (InvalidKeyException e) {
+            return encrypt(bytes, encodingCipher(wrappingString));
+        } catch (InvalidKeySpecException | NoSuchAlgorithmException | InvalidKeyException e) {
             throw new RuntimeException(e);
         } catch (NoSuchPaddingException e) {
             return null;
         }
     }
 
-    public static byte[] unWrapKey(byte[] wrapped, String password) {
+    public static byte[] unwrapByteArrayWithString(byte[] wrapped, String wrappingString) {
         try {
-            Cipher cipher = decodingCipher(password);
+            Cipher cipher = decodingCipher(wrappingString);
             return decrypt(wrapped, cipher);
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
-        } catch (InvalidKeySpecException e) {
-            throw new RuntimeException(e);
-        } catch (InvalidKeyException e) {
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException | InvalidKeyException e) {
             throw new RuntimeException(e);
         } catch (NoSuchPaddingException e) {
             return null;
         }
     }
 
+
+    public static byte[] unWrapKey(byte[] wrapped, String password) {
```

Comment:
I think it's a bit more confusing to have two methods with identical signatures that do the same thing, personally. It makes it seem like one of them must do something the other isn't.